core/spec.md

```diff
@@ -146,7 +146,7 @@ TODO: Add `format`
 - Type: String   # SHOULD this be a URI-Reference?
 - Description: An immutable unique identifier of the entity.
 - Constraints:
-  - MUST be a non-empty string
+  - MUST be a non-empty string consisting of visible US-ASCII octets (33-126).
```

Comment:
Assume we genuinely want all of those characters to be valid, that looks fine to me. I'm not in a position to judge whether we *do* want that character set :)

Reply:
I'm inclined to start with being less restrictive and wait until someone can identify a reason for us to ban more chars.